app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php

```diff
@@ -2,57 +2,113 @@
 
 namespace BodenkoVV\AskQuestion\Block;
 
-use \BodenkoVV\AskQuestion\Model\ResourceModel\Question\CollectionFactory;
-use \BodenkoVV\AskQuestion\Helper\Data;
+use BodenkoVV\AskQuestion\Model\ResourceModel\Question\Collection;
+use BodenkoVV\AskQuestion\Model\ResourceModel\Question\CollectionFactory;
+use BodenkoVV\AskQuestion\Model\QuestionFactory;
+use BodenkoVV\AskQuestion\Helper\Data;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
 
 /**
  * Class TabAskQuestion
  * @package BodenkoVV\AskQuestion\Block
  */
-class TabAskQuestion extends \Magento\Framework\View\Element\Template
+class TabAskQuestion extends Template
 {
-    /**@var \BodenkoVV\AskQuestion\Model\ResourceModel\Question\CollectionFactory      */
+    /**@var CollectionFactory */
     private $collectionFactory;
 
-    /** @var Data $_helperData */
-    public $_helperData;
+    /** @var Data $helperData */
+    public $helperData;
+
+    /** @var QuestionFactory  */
+    public $questionFactory;
+
+    /** @var \Magento\Framework\Registry  */
+    public $_registry;
+
+    /** @var \Magento\Framework\View\Result\PageFactory  */
+    public $resultPageFactory;
+
+    public $askQuestionEnable;
+
+    public $askQuestionText;
+
+    public $askQuestionTitle;
 
-    /**
-     * Requests constructor.
-     * @param CollectionFactory $collectionFactory
-     * @param Data $helperData
-     * @param \Magento\Framework\View\Element\Template\Context $context
-     * @param array $data
-     */
     public function __construct(
+        Context $context,
         CollectionFactory $collectionFactory,
+        QuestionFactory $questionFactory,
         Data $helperData,
-        \Magento\Framework\View\Element\Template\Context $context,
+        \Magento\Framework\Registry $registry,
+        \Magento\Framework\View\Result\PageFactory $resultPageFactory,
         array $data = []
     ) {
         parent::__construct($context, $data);
-        $this->_helperData = $helperData;
+        $this->helperData = $helperData;
         $this->collectionFactory = $collectionFactory;
+        $this->questionFactory = $questionFactory;
+        $this->_registry = $registry;
+        $this->resultPageFactory = $resultPageFactory;
     }
 
     /**
-     * @return object
+     * @return \Magento\Framework\View\Result\Page
      */
-    public function _prepareLayout()
+    public function _toHtml()
     {
-        return parent::_prepareLayout();
+        $this->getQuestionConfig();
+        parent::_toHtml();
+    }
+
+    public function getQuestionConfig()
+    {
+        $resultPage = $this->resultPageFactory->create();
+        if ($this->helperData->getGeneralConfig('bodenkovv_askquestion_enable'))
```

Comment:
This check is not correct. The field bodenkovv_askquestion_enable is not in general section, it's inside of 'cron' section
Please rename this item to "enable_cron" or 'enable_auto_confirming', no need to add vendor name there.
Also I would advice not to use getGeneralConfig, please create more simple function in helper: isCronEnabled() and use the exact constant 'askquestion/cron/enable_auto_confirming'

Reply:
bodenkovv_askquestion_enable in the general section )

Reply:
+